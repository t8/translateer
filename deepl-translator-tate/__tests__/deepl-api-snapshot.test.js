const { getTranslation, splitSentences } = require('../src/deepl-api-helper');

jest.setTimeout(30000);

test('Splits sentences correctly', () => {
  return splitSentences(['Split me, please. Yours truly, the sentence.'], 'EN')
    .then(response => expect(response).toMatchSnapshot())
    .catch(() => fail());
});

test('Gets a correct translation', () => {
  return getTranslation(['Translate me, please.'], 'DE', 'EN')
    .then(response =>
      expect(response).toMatchSnapshot({
        id: expect.any(Number),
        jsonrpc: '2.0',
        result: {
          date: expect.any(String),
          source_lang: 'EN',
          source_lang_is_confident: expect.any(Number),
          target_lang: 'DE',
          timestamp: expect.any(Number),
          translations: [
            {
              beams: [
                {
                  num_symbols: expect.any(Number),
                  postprocessed_sentence: expect.any(String),
                  score: expect.any(Number),
                  totalLogProb: expect.any(Number),
                },
                {
                  num_symbols: expect.any(Number),
                  postprocessed_sentence: expect.any(String),
                  score: expect.any(Number),
                  totalLogProb: expect.any(Number),
                },
                {
                  num_symbols: expect.any(Number),
                  postprocessed_sentence: expect.any(String),
                  score: expect.any(Number),
                  totalLogProb: expect.any(Number),
                },
                {
                  num_symbols: expect.any(Number),
                  postprocessed_sentence: expect.any(String),
                  score: expect.any(Number),
                  totalLogProb: expect.any(Number),
                },
                {
                  num_symbols: expect.any(Number),
                  postprocessed_sentence: expect.any(String),
                  score: expect.any(Number),
                  totalLogProb: expect.any(Number),
                },
              ],
              timeAfterPreprocessing: expect.any(Number),
              timeReceivedFromEndpoint: expect.any(Number),
              timeSentToEndpoint: expect.any(Number),
              total_time_endpoint: expect.any(Number),
            },
          ],
        },
      })
    )
    .catch(() => fail());
});
