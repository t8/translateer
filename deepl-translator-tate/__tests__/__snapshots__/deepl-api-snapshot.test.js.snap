// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gets a correct translation 1`] = `
Object {
  "id": Any<Number>,
  "jsonrpc": "2.0",
  "result": Object {
    "date": Any<String>,
    "source_lang": "EN",
    "source_lang_is_confident": Any<Number>,
    "target_lang": "DE",
    "timestamp": Any<Number>,
    "translations": Array [
      Object {
        "beams": Array [
          Object {
            "num_symbols": Any<Number>,
            "postprocessed_sentence": Any<String>,
            "score": Any<Number>,
            "totalLogProb": Any<Number>,
          },
          Object {
            "num_symbols": Any<Number>,
            "postprocessed_sentence": Any<String>,
            "score": Any<Number>,
            "totalLogProb": Any<Number>,
          },
          Object {
            "num_symbols": Any<Number>,
            "postprocessed_sentence": Any<String>,
            "score": Any<Number>,
            "totalLogProb": Any<Number>,
          },
          Object {
            "num_symbols": Any<Number>,
            "postprocessed_sentence": Any<String>,
            "score": Any<Number>,
            "totalLogProb": Any<Number>,
          },
          Object {
            "num_symbols": Any<Number>,
            "postprocessed_sentence": Any<String>,
            "score": Any<Number>,
            "totalLogProb": Any<Number>,
          },
        ],
        "timeAfterPreprocessing": Any<Number>,
        "timeReceivedFromEndpoint": Any<Number>,
        "timeSentToEndpoint": Any<Number>,
        "total_time_endpoint": Any<Number>,
      },
    ],
  },
}
`;

exports[`Splits sentences correctly 1`] = `
Object {
  "id": 1,
  "jsonrpc": "2.0",
  "result": Object {
    "lang": "EN",
    "lang_is_confident": 1,
    "splitted_texts": Array [
      Array [
        "Split me, please.",
        "Yours truly, the sentence.",
      ],
    ],
  },
}
`;
